src/test/integration/IntegrationChecks.t.sol

```diff
@@ -1,23 +1,65 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity ^0.8.27;
 
-import "src/test/integration/IntegrationBase.t.sol";
+import "src/test/integration/IntegrationUtils.t.sol";
 import "src/test/integration/users/User.t.sol";
 import "src/test/integration/users/User_M1.t.sol";
 import "src/test/integration/users/User_M2.t.sol";
 
 /// @notice Contract that provides utility functions to reuse common test blocks & checks
-contract IntegrationCheckUtils is IntegrationBase {
+contract IntegrationChecks is IntegrationUtils {
     using ArrayLib for *;
-    using SlashingLib for *;
     using StdStyle for *;
 
+    AVS avs;
+    SlashingParams slashParams;
+    SlashingParams slashingParams;
+
+    User operator;
+    OperatorSet operatorSet;
+    AllocateParams allocateParams;
+
+    User staker;
+    IERC20[] tokens;
+    IStrategy[] strategies;
+    uint[] initTokenBalances;
+    uint[] initDepositShares;
+    uint[] withdrawableShares;
+    // Withdrawal[] withdrawals; // cannot copy from memory to storage easily
+    bytes32[] withdrawalRoots;
+    uint[] numTokensRemaining;
+
+    uint64 beaconBalanceGwei;
+    uint40[] validators;
+    uint40[] slashedValidators;
+    uint64 slashedGwei;
+    uint64 slashedAmountGwei;
+
+    function _completeWithdrawal(
+        User staker,
+        User operator,
+        Withdrawal[] memory withdrawals,
+        IStrategy[] memory strategies,
+        uint[] memory withdrawalShares,
+        uint[] memory expectedTokens
+    ) internal {
+        for (uint i = 0; i < withdrawals.length; i++) {
+            if (_randBool()) {
+                staker.completeWithdrawalAsTokens(withdrawals[i]);
+                check_Withdrawal_AsTokens_State(staker, operator, withdrawals[i], withdrawalShares, expectedTokens);
+            } else {
+                staker.completeWithdrawalAsShares(withdrawals[i]);
+                check_Withdrawal_AsShares_State(staker, operator, withdrawals[i], withdrawals[i].strategies, withdrawalShares);
+            }
+        }
+    }
+
     /**
      *
      *                              EIGENPOD CHECKS
      *
      */
-    function check_VerifyWC_State(User staker, uint40[] memory validators, uint64 beaconBalanceGwei) internal {
+    function check_VerifyWC_State(User staker, uint40[] memory validators, uint64 beaconBalanceGwei) public {
```

Comment:
Putting a pin in this, why are these public now?

Reply:
ok i figured this meant tests are calling them like `checker.check_VerifyWC_State()`, but they're not.

could we just keep these internal to keep the diff down, then? maybe im missing where this is needed?